### ISSUE TITLE
DataTable: unfiltered table shows no rows

### ISSUE THREAD

**Issue by MajorPrime** [primefaces/primereact#4107]
### Describe the bug

When the filters object is
```
{
  "null": {
    "operator": "and",
    "constraints": [
      {
        "value": null,
        "matchMode": "startsWith"
      }
    ]
  }
}
```

Then no rows are allowed through the "filter"

I suspect
`localFiltered && localMatch`
should be replaced with
`!localFiltered || localMatch`
on both lines that it occurs

See https://github.com/primefaces/primereact/pull/3820/files which has (correctly) started preventing `localFiltered` from being set to `true`

### Reproducer

https://codesandbox.io/s/primereact-test-forked-wcnnc8

### PrimeReact version

9.2.0

### React version

17.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

**Comment by MajorPrime** [comment]
Occurs when none of the `Column`s have a `field`

**Comment by melloware** [comment]
Do you have a code sandbox reproducer?  You use case does not make sense to me you have no fields and you are trying to filter by "null" ???

**Comment by MajorPrime** [comment]
Added

**Comment by melloware** [comment]
Thanks!

**Comment by ghostd** [comment]
Here is a test to reproduce the case (base on the work of @MajorPrime):
```
import '@testing-library/jest-dom';
import { render } from "@testing-library/react";
import * as React from "react";
import { Column } from '../column/Column';
import { DataTable } from './DataTable';

describe('DataTable', () => {
    test('should display data with Column without field', () => {
        // Arrange
        const { container } = render(<DataTable dataKey={"key"} value={[{ key: 0 }, { key: 2 }]}>
            <Column
                header="Column 1"
                body={(row) => <div>This is column 1 for item {row.key}</div>}
            />
            <Column
                header="Column 2"
                body={(row) => <div>This is column 2 for item {row.key}</div>}
            />
        </DataTable>
        );

        // Assert
        expect(container).toHaveTextContent('This is column 1 for item 0');
        expect(container).toMatchSnapshot();
    });
});

```

**Comment by mertsincan** [comment]
Hi,
I think this issue has already been fixed in 9.2.1;

<img width="914" alt="Screen Shot 2023-03-09 at 16 11 36" src="https://user-images.githubusercontent.com/11868120/224084348-c4f6eec0-c796-49b5-bf6b-0d9af50151eb.png">

Closed for now. If the problem persists, please reopen this ticket. 
Best regards,
